### DISCUSSION TITLE
How to properly set the training precision of a DP model in deepmd-kit v2.1.0

### DISCUSSION THREAD

**Discussion by ipe-zhangyz** [deepmodeling/deepmd-kit#1643] [Q&A] [answered]
Hello, I'm using deepmd-kit 2.1.0. I noticed that there is an environment variable **DP_INTERFACE_PREC** which can be set to "high" or "low", to control high (double) or low (float) precision of training. However, even I set it to "low", I find that the tensors in the descriptor and fitting net are mainly  fp64. I also noticed that there is a training parameter **precision** that can be added in the input.json. The value of his training parameter can be selected among  “default”, “float16”, “float32” and “float64”. If I set it to "float32" , the tensors in the model are mainly fp32. It seems that this training parameter meets my needs more than the **DP_INTERFACE_PREC** env variable, but I don't know if this is a recommended way to control the precision of training. I alse wonder   the relationship of the env variable and the training parameter mentioned above, for example, how does it influence the training if I set **DP_INTERFACE_PREC**  to  "high" and also set the training parameter **precision** to "float32"?

**Comment by njzjz** [comment, marked as answer]
> However, even I set it to "low", I find that the tensors in the descriptor and fitting net are mainly fp64.

I noticed that the default value for `precision` is `float64`, but not `default`. @wanghan-iapcm Shall we change it to `default`?

**Reply by wanghan-iapcm** [reply]
Good idea. We should clearly say that the default precision in descriptors and fitting nets follows the interface precision.

**Reply by ipe-zhangyz** [reply]
So with the current version of deepmd-kit, I should control the training precision using the **precision** parameter in the input.json, no matter what the env variable **DP_INTERFACE_PREC** is. Is this the right information?

**Reply by wanghan-iapcm** [reply]
The `precision` key in descriptor and fitting controls the precision of the numbers in the neural networks, while the environment variable `DP_INTERFACE_PREC` controls the precision of the interface.

**Reply by ipe-zhangyz** [reply]
OK, I get it now. Thank you!
